auth/src/main/java/org/aerogear/android/ags/auth/impl/OIDCTokenAuthenticatorImpl.java

```diff
@@ -1,19 +1,38 @@
 package org.aerogear.android.ags.auth.impl;
 
 import org.aerogear.android.ags.auth.AbstractAuthenticator;
+import org.aerogear.android.ags.auth.AuthService;
+import org.aerogear.android.ags.auth.AuthStateManager;
 import org.aerogear.android.ags.auth.AuthenticationException;
+import org.aerogear.android.ags.auth.IUserPrincipal;
 import org.aerogear.android.ags.auth.credentials.ICredential;
 import org.aerogear.android.ags.auth.credentials.OIDCCredentials;
+import org.aerogear.android.ags.auth.utils.LogoutCompleteHandler;
 import org.aerogear.mobile.core.configuration.ServiceConfiguration;
+import org.aerogear.mobile.core.http.HttpRequest;
+import org.aerogear.mobile.core.http.HttpResponse;
+import org.aerogear.mobile.core.http.HttpServiceModule;
+import org.aerogear.mobile.core.http.OkHttpServiceModule;
 
+import java.net.MalformedURLException;
+import java.net.URL;
 import java.security.Principal;
 
+import static org.aerogear.android.ags.auth.AuthStateManager.getInstance;
+
 /**
  * Authenticates token credentials
  */
 public class OIDCTokenAuthenticatorImpl extends AbstractAuthenticator {
-    public OIDCTokenAuthenticatorImpl(final ServiceConfiguration serviceConfig) {
-        super(serviceConfig);
+
+    private static final String TOKEN_HINT_FRAGMENT = "id_token_hint";
+    private static final String REDIRECT_FRAGMENT = "redirect_uri";
+
+    private final AuthService authService;
+
+    public OIDCTokenAuthenticatorImpl(final ServiceConfiguration config, AuthService authService) {
+        super(config);
+        this.authService = authService;
```

Comment:
I don't think it's necessary to reference the `AuthService` here. We only need the `AuthConfiguration`